packages/theming-core/src/generics/themeRegistry.md

```diff
@@ -0,0 +1,67 @@
+# Generic Theme Registry
+
+The generic theme registry is a helper for creating a directed graph of themes, with the ability to invalidate either subtrees within the hierarchy, or the entire set of themes.  These themes can be queried by name, returning cached values if they are valid, or causing themes to be recreated on demand.
+
+This is set up to be able to support the current Fabric loadTheme/createTheme/getTheme system, while allowing for the addition of named themes when desired.  The dynamic updating of these is a requirement for using a theming library in React Native where actual platform values may change dynamically.
```

Comment:
A named theme is a "scheme" right? How does this related to the schemes work?

Reply:
Our existing scheme work provided by `ThemeProvider` is contextual. Given a scheme name `strong` our schemes will use that relative to the most localized theme, whether that's from a user prop, context, or global theme.

Using schemes as names in a global theme registry is fundamentally different. Scheme names can't simply be `strong` as they are no longer contextual. A `strong` scheme for an inverted sidebar vs. the `strong` scheme for a light-colored document body are fundamentally different themes. For global naming, we will have to ensure that both of these themes have unique names in the global theme registry. Maybe `body-strong` and `sidebar-strong`? The thing I worry about is that it may force components and page sections to be concerned with context they otherwise wouldn't have been concerned about, in this case `body` and `sidebar`.

I'm not sure how much of this is mitigated with `ProcessTheme` which could be a `strong` transform based on the contextual parent. However, I still worry even in this case the theme will require a unique name in the global registry.

Which leads me to another question: Will name collisions be an issue? What happens if multiple components or packages attempt to use `body-string`? Will there be naming conventions such as prepending the package name to the theme name?

Reply:
The concept of storage is fundamentally different between a named registry and existing `Customizer` / `ThemeProvider`, both of which store theme data in `context` (with fallback to global store.)

Reply:
Just brainstorming a bit, I'm wondering if transforms could be registered in a way that would allow web page sections and components to display a scheme relative to a contextual parent (in this case a theme name.)

Could we possibly:
1. Register a transform, such as `registerTransform(schemeName: string, themeProcessor: ProcessTheme)`?
2. Get a theme given a transform and parent (context), such as `getTheme(parent: string, transformName: string) => ITheme`

I'm trying to think of methods of where the contextual approach can be preserved (such as the current theme name in context/props) along with a fixed transform name (`strong`) that aligns with our contextual approach.

This is all assuming of course this isn't addressed in this PR somehow already that I missed. 😄